mypyc/test/test_irbuild.py

```diff
@@ -71,6 +71,9 @@ def run_case(self, testcase: DataDrivenTestCase) -> None:
         if options is None:
             # Skipped test case
             return
+        if "_withgil" in testcase.name and IS_FREE_THREADED:
```

Comment:
Btw, as an option, you can add an opposite marker `_withnogil`, and have two copies of the problematic test with both markers and the different outputs for each marker (I remember we had something similar for daemon, where the order of error messages depends on whether we use cache or not).